src/OldRod.Core/Recompiler/Transform/TypeHelper.cs

```diff
@@ -34,13 +34,13 @@ public class TypeHelper
 
         public TypeHelper(ReferenceImporter importer)
         {
-            var scope = importer.TargetModule.CorLibTypeFactory.CorLibScope;
+            var ownerModule = importer.TargetModule;
+            var factory = ownerModule.CorLibTypeFactory;
+            var scope = ownerModule.CorLibTypeFactory.CorLibScope;
 
-            _arrayType = importer.ImportType(new TypeReference(scope, "System", "Array"));
-            _objectType = importer.ImportType(new TypeReference(scope, "System", "Object"));
+            _arrayType = importer.ImportType(new TypeReference(ownerModule, scope, "System", "Array"));
```

Comment:
Why is this changed?

Reply:
I changed the code to use the constructor which specifies the owner module so that the `Module` property is set. If I remember correctly this was null previously and it is being used on the inside of the TypeReference class and it was causing an issue on a file I tested on. Since I made this change I have a new PC and I don't have the file anymore :/ sorry.

Also, I think the `ImportType` call may be redundant here too as the TypeReference already has a target module, scope, name, and namespace specified.

Even if we aren't going by the bug fix I mentioned above changing the code to set `Module` property and skipping importing would be faster as the importer would not have to process a scope that is already imported and ready to go.

Reply:
My point is (and it applies to all comments), that you should either set the module yourself using the constructor, OR use `ImportType`, not both. Importing a type that is specifically built as an imported reference is redundant.